packages/repo-tools/src/commands/generate-catalog-info/generate-catalog-info.ts

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import YAML from 'js-yaml';
+import pLimit from 'p-limit';
+import { relative as relativePath, resolve as resolvePath } from 'path';
+import { yamlOverwrite } from 'yaml-diff-patch';
+import chalk from 'chalk';
+import { PackageGraph, PackageRole } from '@backstage/cli-node';
+import { Entity } from '@backstage/catalog-model';
+import {
+  getOwnerFromCodeowners,
+  getPossibleCodeowners,
+  loadCodeowners,
+} from './codeowners';
+import {
+  BackstagePackageJson,
+  isBackstagePackage,
+  readFile,
+  writeFile,
+} from './utils';
+import { CodeOwnersEntry } from 'codeowners-utils';
+
+type CreateFixPackageInfoYamlsOptions = {
+  dryRun?: boolean;
+};
+
+export default async (opts: CreateFixPackageInfoYamlsOptions) => {
+  const { dryRun = false } = opts;
+  const packages = await PackageGraph.listTargetPackages();
+  const codeowners = await loadCodeowners();
+  const limit = pLimit(10);
+
+  const results = await Promise.allSettled<void>(
+    packages.map(({ packageJson, dir }) =>
+      limit(async () => {
+        if (!isBackstagePackage(packageJson)) {
+          return;
+        }
+
+        // Check if there is already a corresponding catalog-info.yaml
+        const infoYamlPath = resolvePath(dir, 'catalog-info.yaml');
+        let yamlString = '';
+
+        try {
+          yamlString = await readFile(infoYamlPath, { encoding: 'utf-8' });
+        } catch (e) {
+          if (e.code === 'ENOENT') {
+            await createCatalogInfoYaml({
+              yamlPath: infoYamlPath,
+              packageJson,
+              codeowners,
+              dryRun,
+            });
+            return;
+          }
+
+          throw e;
+        }
+
+        fixCatalogInfoYaml({
```

Comment:
```suggestion
        await fixCatalogInfoYaml({
```

Right? Throwing this in to potentially ship asap :p